docs/publishers/config.rst

```diff
@@ -167,6 +187,22 @@ loads.
 
      This setting can only be set using :js:func:`window.hypothesisConfig`.
 
+   .. option:: onProfileRequest
+
+     ``function``. A JavaScript function that will be called when the user clicks
+     the user profile (user name) link in the sidebar. No arguments are passed
```

Comment:
This PR is getting pretty large so I'm fine with making this a separate issue. I'm pretty sure we will want it though since in eLife's case it provides a way to link annotations to the real life academic identity of the author (who may be eg. the author of the paper).